test_name in test_buffer.py fails on OS X

On OS X `/var` is a symlink to `/private/var` so the check:

```
new_name = vim.eval('tempname()')
vim.current.buffer.name = new_name
eq(vim.current.buffer.name, new_name)
```

fails with:

```
AssertionError: '/private/var/folders/07/jtkf72yn4_587wv8zc7ntg6h0000gn/T/nvim8XLj27/0' != '/var/folders/07/jtkf72yn4_587wv8zc7ntg6h0000gn/T/nvim8XLj27/0'
```

Although the file was name was changed the test fails because the symlink path is being used in one case and the full path in the other case.

--- a/test/test_buffer.py
+++ b/test/test_buffer.py
@@ -101,10 +101,10 @@ def test_number():
 def test_name():
     vim.command('new')
     eq(vim.current.buffer.name, '')
-    new_name = vim.eval('tempname()')
+    new_name = vim.eval('resolve(tempname())')
     vim.current.buffer.name = new_name
     eq(vim.current.buffer.name, new_name)
-    vim.command('w!')
+    vim.command('silent w!')
     ok(os.path.isfile(new_name))
     os.unlink(new_name)
 

